documentation/concepts.md

```diff
@@ -396,6 +396,22 @@ This feature enables complex use cases such as:
 
 This behavior may be enabled by [*configuration*](./configuration.md)
 
+This may be used in tandem with API `FetchPartitionMetadata` to retrieve partition's metadata.
+
+
+## Encryption of data on server side on server side
+
+Streams support the encryption of messages' values on the server side for extra security and data governance concerns.
+
+This behavior may be enabled by [*configuration*](./configuration.md).
+
+For each message, a random AES key is generated and the message is encrypted using that key (called the `Data Key`).
```

Comment:
Is this accurate? My understanding is currently a single data encryption key is generated on partition initialization.

IMO we should encrypt every X messages with the same DEK before changing it. Not totally sure what an appropriate value for X would be.